modules/network/src/main/java/org/apache/ignite/network/internal/recovery/RecoveryClientHandshakeManager.java

```diff
@@ -44,23 +44,29 @@
     /** Handshake completion future. */
     private final CompletableFuture<NettySender> handshakeCompleteFuture = new CompletableFuture<>();
 
+    /** Message factory. */
+    private final NetworkMessagesFactory messageFactory;
+
     /**
      * Constructor.
      *
      * @param launchId Launch id.
      * @param consistentId Consistent id.
      */
-    public RecoveryClientHandshakeManager(UUID launchId, String consistentId) {
+    public RecoveryClientHandshakeManager(
+        UUID launchId, String consistentId, NetworkMessagesFactory messageFactory
+    ) {
         this.launchId = launchId;
         this.consistentId = consistentId;
+        this.messageFactory = messageFactory;
     }
 
     /** {@inheritDoc} */
     @Override public HandshakeAction onMessage(Channel channel, NetworkMessage message) {
         if (message instanceof HandshakeStartMessage) {
             HandshakeStartMessage msg = (HandshakeStartMessage) message;
 
-            HandshakeStartResponseMessage response = HandshakeMessageFactory.handshakeStartResponseMessage()
+            HandshakeStartResponseMessage response = messageFactory.handshakeStartResponseMessage()
```

Comment:
BTW, why can't these methods be static?

Reply:
having factories with only static methods kinda diminishes the whole idea of having factories at all....

Reply:
We can discuss it with other guys. People name too many different things as "factories". This one is not factory in a traditional meaning. It's more like "Replace Constructor with Factory Method" pattern applied to every builder.

Reply:
I strongly disagree with this approach. Having static factories is similar to having singletons, not a very appreciated design. Having factories, that are injected, is a more flexible approach while being not very verbose compared to the static factories.